### ISSUE TITLE
Error when upgrading using Helm from 0.8.0 to 1.0.0

### ISSUE THREAD

**Issue by ymesika** [istio/istio#7023]
I first installed v0.8.0 and then tried an upgrade to 1.0 using `release-1.0` with:
```sh
helm upgrade istio install/kubernetes/helm/istio --namespace istio-system
```

An error is being printed  by Helm:
```sh
Error: UPGRADE FAILED: ClusterRoleBinding.rbac.authorization.k8s.io "istio-ingress-istio-system" is invalid: roleRef: Invalid value: rbac.RoleRef{APIGroup:"rbac.authorization.k8s.io", Kind:"ClusterRole", Name:"istio-ingress-istio-system"}: cannot change roleRef
```

**Comment by mandarjog** [comment]
This was the error that I was worried about in the config.istio.io -> rbac.istio.io change.

Roleref immutable.

**Comment by hklai** [comment]
@ymesika are you fixing it?

**Comment by costinm** [comment]
I suggest rolling back the rename - we decided to not rename networking from alpha3 at this point, and do all renames with K8S 1.11 is available.

**Comment by ymesika** [comment]
Guys, it's not related to the rename PR which didn't even get merged yet.
Just reported what I'm seeing when I upgrade to `release-1.0` branch.

**Comment by ymesika** [comment]
I can confirm that partially reverting #6390 and changing:

```yaml
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: istio-ingress-{{ .Release.Namespace }}
```
back to:

```yaml
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: istio-pilot-{{ .Release.Namespace }}
```
solves this and upgrade to current `release-1.0` works fine.

It's a fix which brings back a typo but solves this issue.

@costinm @hklai your call whether to put the typo back.

**Comment by costinm** [comment]
Ok, let's not merge it then :-) 

I remember something similar in 0.8 - it happens if any labels get changed. It might be related to the 0.8 labels that got converted to 1.0.

**Comment by costinm** [comment]
Yes please, put the typo back.

**Comment by ymesika** [comment]
Fixed by #7062
